src/notification/index.js

```diff
@@ -260,8 +257,24 @@ export class NotificationListener {
 
   /** Start listening.  Don't call twice without intervening `stop`. */
   start() {
-    this.listen('notificationOpened', this.handleNotificationOpen);
+    if (Platform.OS === 'android') {
+      // On Android, the object passed to the handler is constructed in
+      // FcmMessage.kt, and will always be a Notification.
+      this.listen('notificationOpened', this.handleNotificationOpen);
+    } else {
+      // On iOS, `note` should be an IOSNotifications object. The notification
+      // data it returns from `getData` is unvalidated -- it comes almost
+      // straight off the wire from the server.
+      this.listen('notificationOpened', (note: NotificationMuddle) => {
+        const data = extractNotificationData(note);
+        if (data) {
+          this.handleNotificationOpen(data);
+        }
```

Comment:
No, there's a very big change here – messages from the Android side are no longer being run through `extractNotificationData`.

On the iOS side, though, it is supposed to be `[nfc]` (though that could be explained better). The conditional in question is effectively lifted from `narrowToNotification`, which is the external destination of this value; it cannot be removed from this commit without replacing the `$FlowFixMe` deleted from `handleNotificationOpen`.

Part of the point of this change is to make `handleNotificationOpen` properly typed, internally and externally, and to take precisely a `Notification` rather than a `NotificationMuddle`.

Reply:
> No, there's a very big change here – messages from the Android side are no longer being run through `extractNotificationData`.

Hmm, but I think they already weren't:
```
-  handleNotificationOpen = (notification: NotificationMuddle) => {
-    const data: Notification =
-      // $FlowFixMe clarify notification types
-      Platform.OS === 'ios' ? extractNotificationData(notification) : notification;
-    this.dispatch(narrowToNotification(data));
```
In the Android case, that doesn't call `extractNotificationData`.

Reply:
> The conditional in question is effectively lifted from `narrowToNotification`, which is the external destination of this value; it cannot be removed from this commit without replacing the `$FlowFixMe` deleted from `handleNotificationOpen`.

Deleting fixmes is a great thing!

Even better, though, is deleting them in a minimal change, before and/or after NFC commits that make related changes. It looks like that should be possible here.

Reply:
> In the Android case, that doesn't call `extractNotificationData`.

:facepalm: Right. Somehow completely missed that on my first *and* second pass. Will adjust.